### ISSUE TITLE
tau_psy.f90 added to build

### ISSUE THREAD

**Issue by hiker** [MetOffice/lfric-baf#6]
I am surprised that we don't see this problem on NCI. When I install lfric-baf, it copies the `tau_psy.f90` files into the infrastructure files, which means this file will be compiled (which then results in linking errors).

1. Any idea why we don't see this on NCI? I need to manually remove this file from infrastructure/soure/psydata.
2. Depending on the answer for 1. above, it might be better to move this file out of the infrastructure/source tree, add it to infrastructure/build/psyclone (which doesn't exist here yet, but it is in LFRic core). And then if requested (i.e. tau selected), then grab that file? 

The latter approach would also make it easier to support different profiling libs (e.g. Vernier), where atm I am manually copying the psyclone wrapper file into the sources. We could all there wrapper to the psyclone directories, and let fab copy the right file.

**Comment by hiker** [comment]
I just realised that on NCI atm we don't trigger any profiling. So no psy_data calls are added to the code, so the fab dependency analysis will never include this file to be compiled.

Still, it would be a better solution if baf grabs up the wrapper files explicitly (depending on command line options).

**Comment by hiker** [comment]
Fixed in e5e82de7b75f97ca2f3f1f396a516d8c73dd7270 (part of vernier branch).

**Comment by hiker** [comment]
Can this issue actually be closed? @jasonjunweilyu , do we have any CI that runs tau with fab to verify that this works?

**Comment by jasonjunweilyu** [comment]
> Can this issue actually be closed? [@jasonjunweilyu](https://github.com/jasonjunweilyu) , do we have any CI that runs tau with fab to verify that this works?

Yes, I am closing this right now. I just created a pipeline for testing tau as the linker with fab and gungho_model. The build and run are successful (Build: https://git.nci.org.au/bom/ngm/lfric/lfric_atm-fab/-/jobs/83633; Run: https://git.nci.org.au/bom/ngm/lfric/lfric_atm-fab/-/jobs/83634). Profiling information is complete for the kernels.
